processing/src/main/java/org/apache/druid/query/QueryContext.java

```diff
@@ -168,7 +168,20 @@ public Object getOrDefault(String key, Object defaultValue)
   @Nullable
   public String getAsString(String key)
   {
-    return (String) get(key);
+    Object val = get(key);
+    return val == null ? null : val.toString();
+  }
+
+  public String getAsString(String key, String defaultValue)
+  {
+    Object val = get(key);
+    return val == null ? defaultValue : val.toString();
+  }
+
+  @Nullable
+  public Boolean getAsBoolean(String parameter)
```

Comment:
Nit: best to not call query context values "parameters" since a query can have "parameters" which are not in the context. Suggestion: "key".